config/confighttp/confighttp.go

```diff
@@ -213,13 +214,21 @@ func (hss *HTTPServerSettings) ToServer(handler http.Handler, settings component
 	}
 	// TODO: emit a warning when non-empty CorsHeaders and empty CorsOrigins.
 
+	tp := settings.TracerProvider
+	if tp == nil {
+		tp = otel.GetTracerProvider()
```

Comment:
Should this use the global `TracerProvider`, or a `NoopTracerProvider`?  I can see arguments for either and I'm not sure which is preferable here.

The same applies for `MeterProvider` below.